File: src/main/java/no/nav/domain/pensjon/regler/grunnlag/GenerellHistorikk.java
package no.nav.domain.pensjon.regler.grunnlag;

import no.nav.domain.pensjon.regler.beregning.Sertillegg;
import no.nav.domain.pensjon.regler.kode.Fravik_19_3Cti;
import no.nav.domain.pensjon.regler.kode.PoengtilleggCti;

import java.io.Serializable;

/**
 * Generell historisk info for en bruker
 */
public class GenerellHistorikk implements Serializable {

    /**
     * Identifikasjon av den generelle historikken
     */
    private Long generellHistorikkId;
    /*
     * Koder som sier hvilken trygdetid som skal brukes i beregning når første virk er før 1991.
     * Brukes i utenlandssaker der pensjonisten kan få godskrevet trygdetid opptjent i utlandet før 1967.
     */
    private Fravik_19_3Cti fravik_19_3;
    /*
     * Gjennomsnittlig uføregrad - EØS.
     */
    private Double fpp_eos;
    /**
     * Ventetilleggsgrunnlag
     */
    private Ventetilleggsgrunnlag ventetilleggsgrunnlag;

    private PoengtilleggCti poengtillegg;

    /*
     * Inneholder informasjon ang tidligere EØS beregninger. Brukes ved konvertering til AP.
     */
    private EosEkstra eosEkstra;

    /**
     * Innholder trygdetidsgarantien for ektefeller som går under gammel lov før 1.1.1991
     */
    private GarantiTrygdetid garantiTrygdetid;

    /* CR175446 - 1943 konvertert AP1967 til AP2011 */
    /* Særtillegget brukeren hadde på AP1967-ytelsen (settes lik null om han ikke hadde særtillegg) */
    private Sertillegg sertillegg1943kull;

    /* CR175446 - Gift eller tilsvarende med samme person siden 31.12.2010 */
    private boolean giftFor2011;

}

File: src/main/kotlin/no/nav/pensjon/regler/domain/grunnlag/GenerellHistorikk.kt
package no.nav.pensjon.regler.domain.grunnlag

import no.nav.pensjon.regler.domain.beregning.Sertillegg
import no.nav.pensjon.regler.domain.enum.Fravik_19_3_Enum
import no.nav.pensjon.regler.domain.enum.PoengtilleggEnum
import no.nav.pensjon.regler.domain.kode.Fravik_19_3Cti
import no.nav.pensjon.regler.domain.kode.PoengtilleggCti
import java.io.Serializable

/**
 * Generell historisk info for en bruker
 */
class GenerellHistorikk : Serializable {
    /**
     * Identifikasjon av den generelle historikken
     */
    var generellHistorikkId: Long? = null

    /*
     * Koder som sier hvilken trygdetid som skal brukes i beregning når Første virk er før 1991.
     * Brukes i utenlandssaker der pensjonisten kan få godskrevet trygdetid opptjent i utlandet før 1967.
     */
    var fravik_19_3: Fravik_19_3Cti? = null
    var fravik_19_3Enum: Fravik_19_3_Enum? = null

    /*
     * Gjennomsnittlig Uføregrad - EØS.
     */
    var fpp_eos: Double? = null

    /**
     * Ventetilleggsgrunnlag
     */
    var ventetilleggsgrunnlag: Ventetilleggsgrunnlag? = null
    var poengtillegg: PoengtilleggCti? = null
    var poengtilleggEnum: PoengtilleggEnum? = null

    /*
     * Inneholder informasjon ang tidligere EØS beregninger. Brukes ved konvertering til AP.
     */
    var eosEkstra: EosEkstra? = null

    /**
     * Innholder trygdetidsgarantien for ektefeller som går under gammel lov før 1.1.1991
     */
    var garantiTrygdetid: GarantiTrygdetid? = null

    /* CR175446 - 1943 konvertert AP1967 til AP2011 */ /* særtillegget brukeren hadde på AP1967-ytelsen (settes lik null om han ikke hadde særtillegg) */
    var sertillegg1943kull: Sertillegg? = null

    /* CR175446 - Gift eller tilsvarende med samme person siden 31.12.2010 */
    var giftFor2011 = false

}